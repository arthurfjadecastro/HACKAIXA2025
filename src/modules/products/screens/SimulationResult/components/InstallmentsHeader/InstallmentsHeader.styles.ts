import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  installmentsHeader: {
    paddingHorizontal: 0,
    paddingTop: 32,
    paddingBottom: 16,
  },

  installmentsTitle: {
    fontSize: 18,
    fontWeight: '800',
    color: '#1565C0',
    textAlign: 'left',
  },
});
