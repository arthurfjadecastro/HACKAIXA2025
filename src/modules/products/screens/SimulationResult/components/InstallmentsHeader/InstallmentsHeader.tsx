import React from 'react';
import { View } from 'react-native';
import { Text } from '@/design-system/components';
import { styles } from './InstallmentsHeader.styles';

export const InstallmentsHeader: React.FC = () => {
  return (
    <View style={styles.installmentsHeader}>
      <Text style={styles.installmentsTitle}>Parcelas</Text>
    </View>
  );
};
