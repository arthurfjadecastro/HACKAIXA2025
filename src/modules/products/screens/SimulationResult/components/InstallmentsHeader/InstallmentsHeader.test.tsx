import React from 'react';
import { render } from '@testing-library/react-native';
import { InstallmentsHeader } from './InstallmentsHeader';

// Mock do design-system components
jest.mock('@/design-system/components', () => ({
  Text: ({ children, style, ...props }: any) => {
    const { Text: RNText } = require('react-native');
    return <RNText style={style} {...props}>{children}</RNText>;
  },
}));

describe('InstallmentsHeader', () => {
  it('should render correctly', () => {
    const { getByText } = render(<InstallmentsHeader />);
    
    expect(getByText('Parcelas')).toBeTruthy();
  });

  it('should render with correct structure', () => {
    const { getByText } = render(<InstallmentsHeader />);
    
    const parcelasText = getByText('Parcelas');
    expect(parcelasText).toBeTruthy();
  });

  it('should apply correct styles to container', () => {
    const { getByText } = render(<InstallmentsHeader />);
    
    const parcelasText = getByText('Parcelas');
    const container = parcelasText.parent;
    
    expect(container).toBeTruthy();
  });

  it('should render Text component with correct content', () => {
    const { getByText } = render(<InstallmentsHeader />);
    
    const titleElement = getByText('Parcelas');
    expect(titleElement).toBeTruthy();
    expect(titleElement.props.children).toBe('Parcelas');
  });

  it('should be accessible', () => {
    const { getByText } = render(<InstallmentsHeader />);
    
    const titleElement = getByText('Parcelas');
    expect(titleElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const component = render(<InstallmentsHeader />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should have single child text element', () => {
    const { getByText } = render(<InstallmentsHeader />);
    
    const parcelasText = getByText('Parcelas');
    expect(parcelasText).toBeTruthy();
  });

  it('should render without props', () => {
    expect(() => render(<InstallmentsHeader />)).not.toThrow();
  });

  it('should have correct component structure', () => {
    const { getByText } = render(<InstallmentsHeader />);
    
    const parcelasText = getByText('Parcelas');
    expect(parcelasText).toBeTruthy();
  });

  it('should display installments header title', () => {
    const { getByText } = render(<InstallmentsHeader />);
    
    const titleText = getByText('Parcelas');
    expect(titleText).toBeTruthy();
    expect(titleText.props.children).toBe('Parcelas');
  });
});
