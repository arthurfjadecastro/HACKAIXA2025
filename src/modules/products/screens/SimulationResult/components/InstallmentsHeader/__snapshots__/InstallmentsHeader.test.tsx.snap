// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallmentsHeader should match snapshot 1`] = `
<View
  style={
    {
      "paddingBottom": 16,
      "paddingHorizontal": 0,
      "paddingTop": 32,
    }
  }
>
  <Text
    style={
      {
        "color": "#1565C0",
        "fontSize": 18,
        "fontWeight": "800",
        "textAlign": "left",
      }
    }
  >
    Parcelas
  </Text>
</View>
`;
